tableauserverclient/server/endpoint/datasources_endpoint.py

```diff
@@ -190,6 +194,12 @@ def publish(self, datasource_item, file_path, mode, connection_credentials=None)
                                                                               file_contents,
                                                                               connection_credentials)
         server_response = self.post_request(url, xml_request, content_type)
-        new_datasource = DatasourceItem.from_response(server_response.content, self.parent_srv.namespace)[0]
-        logger.info('Published {0} (ID: {1})'.format(filename, new_datasource.id))
-        return new_datasource
+
+        if as_job:
```

Comment:
This can come as a second PR but it'd be great if we actually tracked the progress of the job, or at least optionally tracked the progress of the job, and printed or reported it somehow.